### PR TITLE
[WIP] Cleanup template parameters

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-CSharp/appsettings.json [hunk 1]
```diff
@@ -1,22 +1,6 @@
...
-//#if (!SingleOrgAuth)
-//    "Instance": "https:////login.microsoftonline.com/common",
-//#else
-//    "Instance": "https:////login.microsoftonline.com/",
-//    "Domain": "qualified.domain.name",
-//    "TenantId": "22222222-2222-2222-2222-222222222222",
-//#endif
-//    "ClientId": "11111111-1111-1111-11111111111111111"
-//  },
+//#if(ActiveDirectoryAuth)
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-CSharp/appsettings.json]: AzureActiveDirectoryAuth

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/appsettings.json [hunk 2]
```diff
@@ -1,26 +1,6 @@
...
-//    "Instance": "https:////login.microsoftonline.com/common",
-//#elseif (SingleOrgAuth)
-//    "Instance": "https:////login.microsoftonline.com/",
-//    "Domain": "qualified.domain.name",
-//    "TenantId": "22222222-2222-2222-2222-222222222222",
-//#endif
-//    "ClientId": "11111111-1111-1111-11111111111111111",
-//    "CallbackPath": "/signin-oidc"
-//  },
+//#if(ActiveDirectoryAuth)
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/appsettings.json]: AzureActiveDirectoryAuth

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/appsettings.json [hunk 3]
```diff
@@ -1,26 +1,6 @@
...
-//    "Instance": "https:////login.microsoftonline.com/common",
-//#elseif (SingleOrgAuth)
-//    "Instance": "https:////login.microsoftonline.com/",
-//    "Domain": "qualified.domain.name",
-//    "TenantId": "22222222-2222-2222-2222-222222222222",
-//#endif
-//    "ClientId": "11111111-1111-1111-11111111111111111",
-//    "CallbackPath": "/signin-oidc"
-//  },
+//#if(ActiveDirectoryAuth)
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/appsettings.json]: AzureActiveDirectoryAuth

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/.template.config/vs-2017.3.host.json [hunk 4]
```diff
@@ -25,24 +25,115 @@
...
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "TenantId": "TenantId",
+            "CallbackPath": "CallbackPath"
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/.template.config/vs-2017.3.host.json]: Could you avoid passing CallbackPath as a parameter for OrgAuth SSO and MMO only? it appears that we were not passing it explicitly before, instead the templates were just using their default value specified in the template.json. I'll get this fixed for preview 4

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/.template.config/vs-2017.3.host.json [hunk 5]
```diff
@@ -25,24 +25,115 @@
...
+            },
+            {
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "CallbackPath": "CallbackPath"
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/.template.config/vs-2017.3.host.json]: Here as well

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/vs-2017.3.host.json [hunk 6]
```diff
@@ -25,24 +25,113 @@
...
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "TenantId": "TenantId",
+            "CallbackPath": "CallbackPath"
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/vs-2017.3.host.json]: Avoid this one for now.

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/vs-2017.3.host.json [hunk 7]
```diff
@@ -25,24 +25,113 @@
...
+            },
+            {
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "CallbackPath": "CallbackPath"
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/vs-2017.3.host.json]: and here,

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-CSharp/.template.config/vs-2017.3.host.json [hunk 8]
```diff
@@ -26,12 +26,73 @@
...
+              "name": "internalAuth",
+              "destinationSymbol": "internalAuth"
+            }
+
+          ],
+          "parameters": {
+            "Domain": "Domain",
+            "ClientId": "ClientId",
+            "TenantId": "TenantId",
+            "CallbackPath": "CallbackPath",
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-CSharp/.template.config/vs-2017.3.host.json]: Avoid this one

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/template.json [hunk 9]
```diff
@@ -278,21 +226,33 @@
       "type": "computed",
       "value": "(auth == \"Individual\")"
     },
-    "IndividualAuth": {
-      "type": "computed",
-      "value": "(auth == \"Individual\" || auth == \"IndividualB2C\")"
-    },
     "IndividualB2CAuth": {
       "type": "computed",
       "value": "(auth == \"IndividualB2C\")"
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/template.json]: internalAuth

Comment (phenning) [reply src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/template.json]: You also need to update SingleOrgAuth and MultiOrgAuth computed symbols as well, like was done with Razor template

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-CSharp/.template.config/template.json [hunk 10]
```diff
@@ -182,21 +160,33 @@
       "type": "computed",
       "value": "(auth == \"SingleOrg\")"
     },
-    "IndividualAuth": {
+    "IndividualB2CAuth": {
       "type": "computed",
       "value": "(auth == \"IndividualB2C\")"
```
Comment (phenning) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-CSharp/.template.config/template.json]: internalAuth, as well as fix SingleOrgAuth and MulitOrg auth as well.

**Comment by javiercn** [comment]
Closing this as we are punting it

**Pull request closed without merging**